storage/block_storage.go

```diff
@@ -223,18 +403,14 @@ func (b *BlockStorage) GetBlockLazy(
 	return b.getBlockResponse(ctx, blockIdentifier, transaction)
 }
 
-// GetBlock returns a block, if it exists. GetBlock
-// will fetch all transactions contained in a block
-// automatically. If you don't wish to do this for
-// performance reasons, use GetBlockLazy.
-func (b *BlockStorage) GetBlock(
+// GetBlockTransactional gets a block in the context of a database
+// transaction.
+func (b *BlockStorage) GetBlockTransactional(
```

Comment:
The names for the functions `GetBlock` and `GetBlockTransactional` feel like they should be switched to me, or renamed... it's somewhat confusing because `GetBlock` is actually the function invoking a `NewDatabaseTransaction`

Reply:
Ahh I consider the naming from the perspective of the caller. With `*Transactional`, the caller is responsible for providing the `DatabaseTransaction` to use. Without `Transactional`, the caller is not responsible for providing a transaction where the query will be performed.